### PR TITLE
Actual grpc trigger

### PR DESCRIPTION
**Pull request by schoren**
This PR adds gRPC triggering capabilites.

## Changes

-

## Fixes

-

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: server/executor/trigger/grpc.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+	options := grpcurl.FormatOptions{
+		EmitJSONDefaultFields: true,
+		IncludeTextSeparator:  true,
+		AllowUnknownFields:    true,
+	}
+
+	rf, _, err := grpcurl.RequestParserAndFormatter(grpcurl.FormatJSON, desc, strings.NewReader(tReq.Request), options)
+	if err != nil {
+		return response, fmt.Errorf("failed to construct request parser and formatter for %q", err)
```
Comment (mathnogueira) [inline server/executor/trigger/grpc.go:82]: I think you are either using the wrong variable or the wrong format option:

Either it's `%q` but the variable is not `err` or it should be `%w` and the variable is `error` indeed.

**Review by mathnogueira — APPROVED** [review]
LGTM! Just two small things

File: server/executor/trigger/grpc.go (hunk 1 above)
Comment (schoren) [reply server/executor/trigger/grpc.go:82]: good catch!

**Comment by mathnogueira** [comment]
@schoren feel free to merge it

**Pull request merged**